clients/wallet/src/cache.rs

```diff
@@ -158,7 +158,7 @@ impl WalletStateStorage {
 	/// Removes the directory itself.
 	/// User should not be able to do this in production.
 	pub fn remove_dir(&self) -> bool {
-		std::fs::remove_dir_all(&self.path).is_ok()
+		std::fs::remove_dir_all(&self.path).is_ok() || !Path::new(&self.path).is_dir()
```

Comment:
Is this just used to rturn a proper boolean in case the directory did not exist or am I misreading this?

Reply:
updated the method to return nothing, and perform the `remove_dir_all()` only if the path exists.